maintenance/migratestatus/migrator/migrator.go

```diff
@@ -0,0 +1,281 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package migrator
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/google/go-github/github"
+	"k8s.io/test-infra/maintenance/githubutil"
+)
+
+var (
+	stateAny     = "ANY_STATE"
+	stateSuccess = "success"
+
+	deprecatedDesc = "This status context is deprecated."
+)
+
+// contextCondition is a struct that describes a condition about the state or existance of a context.
+type contextCondition struct {
+	// context is the status context that this condition applies to.
+	context string
+	// state is the status state that the condition accepts, "ANY_STATE", or nil to accept only if
+	// the status is not present.
+	state *string
```

Comment:
I was using *string before I added the ANY_STATE value because it provided nil as an out of band value. Since I'm using ANY_STATE now it definitely makes more sense to replace nil with a magic value as well. Done.